airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/ContainerOrchestratorApp.java

```diff
@@ -54,108 +38,57 @@
-    System.out.println(Jsons.serialize(replicationOutput));
-
-    LOGGER.info("Replication runner complete!");
-  }
-
   public static void main(final String[] args) throws Exception {
     WorkerHeartbeatServer heartbeatServer = null;
 
     try {
       // read files that contain all necessary configuration
-      final String application = Files.readString(Path.of(ReplicationLauncherWorker.INIT_FILE_APPLICATION));
+      final String application = Files.readString(Path.of(OrchestratorConstants.INIT_FILE_APPLICATION));
       final Map<String, String> envMap =
-          (Map<String, String>) Jsons.deserialize(Files.readString(Path.of(ReplicationLauncherWorker.INIT_FILE_ENV_MAP)), Map.class);
+          (Map<String, String>) Jsons.deserialize(Files.readString(Path.of(OrchestratorConstants.INIT_FILE_ENV_MAP)), Map.class);
 
       final Configs configs = new EnvConfigs(envMap::get);
 
       heartbeatServer = new WorkerHeartbeatServer(WorkerApp.KUBE_HEARTBEAT_PORT);
       heartbeatServer.startBackground();
 
-      if (application.equals(ReplicationLauncherWorker.REPLICATION)) {
-        replicationRunner(configs);
-      } else {
-        LOGGER.error("Runner failed", new IllegalStateException("Unexpected value: " + application));
-        System.exit(1);
-      }
+      final WorkerConfigs workerConfigs = new WorkerConfigs(configs);
+      final ProcessFactory processFactory = getProcessBuilderFactory(configs, workerConfigs);
+      final JobOrchestrator<?> jobOrchestrator = getJobOrchestrator(configs, workerConfigs, processFactory, application);
+
+      log.info("Starting {} orchestrator...", jobOrchestrator.getOrchestratorName());
+      jobOrchestrator.runJob();
+      log.info("{} orchestrator complete!", jobOrchestrator.getOrchestratorName());
     } finally {
       if (heartbeatServer != null) {
-        LOGGER.info("Shutting down heartbeat server...");
+        log.info("Shutting down heartbeat server...");
         heartbeatServer.stop();
       }
     }
 
     // required to kill kube client
-    LOGGER.info("Runner closing...");
+    log.info("Runner closing...");
     System.exit(0);
   }
 
+  private static JobOrchestrator<?> getJobOrchestrator(final Configs configs,
+                                                       final WorkerConfigs workerConfigs,
+                                                       final ProcessFactory processFactory,
+                                                       final String application) {
+    if (application.equals(ReplicationLauncherWorker.REPLICATION)) {
+      return new ReplicationJobOrchestrator(configs, workerConfigs, processFactory);
+    } else if (application.equals(NormalizationLauncherWorker.NORMALIZATION)) {
+      return new NormalizationJobOrchestrator(configs, workerConfigs, processFactory);
+    } else {
+      log.error("Runner failed", new IllegalStateException("Unexpected value: " + application));
+      System.exit(1);
```

Comment:
why do we need to `exit` here?

Reply:
In case there's any problem with killing logging/heartbeat servers/etc I want to guarantee exiting so there's no chance it hangs.